Add GraphQL API reference (#35)
vtex-apps/tax-protocol-example: docs/tax-protocol-graphql-api.md

# Tax Protocol GraphQL API

This API enables setting and retrieving the Tax Configuration, which is part of the [OrderForm configuration](https://developers.vtex.com/docs/guides/get-an-account-orderform-configuration), for the Tax Service.

> ## Schema Overview

Reviewer: ```suggestion
## Schema overview
```

> ## Query

Reviewer: ```suggestion
### Query
```

```mermaid
classDiagram
  direction LR
  class TaxConfiguration {
    allowExecutionAfterErrors Boolean
    authorizationHeader String
    integratedAuthentication Boolean
    url String
  }

  class Query {
    getTaxConfiguration TaxConfiguration
  }

  Query --> TaxConfiguration : getTaxConfiguration

  click Query href "#query"
  click TaxConfiguration href "#taxconfiguration"
```

<table>
  <thead>
    <tr>
      <th align="left">Field</th>
      <th align="right">Argument</th>
      <th align="left">Type</th>
      <th align="left">Description</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td colspan="2" valign="top">
        <strong id="query.gettaxconfiguration">getTaxConfiguration</strong>
      </td>
      <td valign="top"><a href="#taxconfiguration">TaxConfiguration</a></td>

>       <td>Endpoint responsible to fetch information regarding the Tax Configuration of an account. Its resolver makes a GET request to <code>Checkout</code> (<a href="https://developers.vtex.com/docs/api-reference/checkout-api#get-/api/checkout/pvt/configuration/orderForm" target="_blank">Get orderForm configuration</a>) and retrieves the Tax Configuration from the OrderForm Configuration.</td>

Reviewer: ```suggestion
      <td>Retrieves the tax configuration for an account from the orderForm using the Checkout API (<a href="https://developers.vtex.com/docs/api-reference/checkout-api#get-/api/checkout/pvt/configuration/orderForm" target="_blank">Get orderForm configuration</a>).</td>
```

    apps [App]
    allowMultipleDeliveries Boolean
    allowManualPrice Boolean
    maxIntOfWhiteLabelSellers Int
    maskFirstPurchaseData Boolean
    recaptchaValidation Boolean
  }

  class Mutation {
    setTaxConfiguration(operation String) OrderFormConfiguration
  }

  OrderFormConfiguration --> PaymentConfiguration : paymentConfiguration
  OrderFormConfiguration --> TaxConfiguration : taxConfiguration
  OrderFormConfiguration --> App : apps
  Mutation --> OrderFormConfiguration : setTaxConfiguration

  click Mutation href "#mutation"
  click OrderFormConfiguration href "#orderformconfiguration"
  click PaymentConfiguration href "#paymentconfiguration"
  click TaxConfiguration href "#taxconfiguration"
  click App href "#app"
```

<table>
  <thead>
    <tr>
      <th align="left">Field</th>
      <th align="right">Argument</th>
      <th align="left">Type</th>
      <th align="left">Description</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td colspan="2" valign="top">
        <strong id="mutation.settaxconfiguration">setTaxConfiguration</strong>
      </td>
      <td valign="top">
        <a href="#orderformconfiguration">OrderFormConfiguration</a>
      </td>

>       <td>Endpoint responsible for activating and deactivating the tax service on the <a href="https://developers.vtex.com/docs/guides/update-an-account-orderform-configuration" target="_blank">OrderForm configuration</a>. It does a POST HTTP request (<a href="https://developers.vtex.com/docs/api-reference/checkout-api#post-/api/checkout/pvt/configuration/orderForm" target="_blank">Update orderForm configuration</a>) to set the Tax Configuration on the OrderForm Configuration. In order to do so, it uses a Checkout <a href="https://developers.vtex.com/docs/guides/vtex-io-documentation-clients" target="_blank">client</a>.</td>

Reviewer: ```suggestion
      <td>Activates or deactivates the tax service in the <a href="https://developers.vtex.com/docs/guides/update-an-account-orderform-configuration" target="_blank">orderForm configuration</a> using the (<a href="https://developers.vtex.com/docs/api-reference/checkout-api#post-/api/checkout/pvt/configuration/orderForm" target="_blank">Update orderForm configuration</a>) endpoint.</td>
```

    </tr>
    <tr>
      <td colspan="2" align="right" valign="top">operation</td>
      <td valign="top">String</td>
      <td>Argument to define whether the tax service will be activated or deactivated. Possible values: <code>"activate"</code>, <code>"deactivate"</code>.</td>
    </tr>
  </tbody>
</table>

> ## Schema Description

Reviewer: ```suggestion
## Schema description
```

> >⚠️ Arguments must be provided by the user. The exclamation mark `!` indicates that a type cannot be nullable.

Reviewer: ```suggestion
>⚠️ Arguments must be provided by the user. Required fields are marked with an exclamation mark (`!`).
```

  </thead>
  <tbody>
    <tr>
      <td colspan="2" valign="top">
        <strong id="orderformconfiguration.paymentconfiguration">paymentConfiguration</strong>
      </td>
      <td valign="top">
        <a href="#paymentconfiguration">PaymentConfiguration</a>
      </td>
      <td>Payment configuration information.</td>
    </tr>
    <tr>
      <td colspan="2" valign="top">
        <strong id="orderformconfiguration.taxconfiguration">taxConfiguration</strong>
      </td>
      <td valign="top">
        <a href="#taxconfiguration">TaxConfiguration</a>
      </td>
      <td>External tax service configuration information.</td>
    </tr>
    <tr>
      <td colspan="2" valign="top">
        <strong id="orderformconfiguration.minimumquantityaccumulatedforitems">
          minimumQuantityAccumulatedForItems</strong>
      </td>
      <td valign="top">
        Int
      </td>
      <td>Minimum SKU quantity by cart. This field is mandatory and must be sent in all requests.</td>
    </tr>
    <tr>
      <td colspan="2" valign="top">
        <strong id="orderformconfiguration.decimaldigitsprecision">decimalDigitsPrecision</strong>
      </td>
      <td valign="top">
        Int
      </td>

>       <td>Number of price digits.</td>

Reviewer: ```suggestion
      <td>Number of decimal price digits.</td>
```

    </tr>
    <tr>
      <td colspan="2" valign="top">
        <strong id="orderformconfiguration.minimumvalueaccumulated">minimumValueAccumulated</strong>
      </td>
      <td valign="top">
        Int
      </td>
      <td>Minimum cart value.</td>
    </tr>
    <tr>
      <td colspan="2" valign="top">
        <strong id="orderformconfiguration.apps">apps</strong>
      </td>
      <td valign="top">[<a href="#app">App</a>]</td>
      <td>List of App objects, where each object has an app configuration information. See an example with the <code>apps</code> list in the <a href="https://developers.vtex.com/docs/guides/get-an-account-orderform-configuration" target="_blank">Get orderForm configuration</a> response body.</td>
    </tr>
    <tr>
      <td colspan="2" valign="top">
        <strong id="orderformconfiguration.allowmultipledeliveries">allowMultipleDeliveries</strong>
      </td>
      <td valign="top">
        Boolean
      </td>

>       <td>Allows the selection of items from several delivery channels in the same purchase.</td>

Reviewer: ```suggestion
      <td>Enables multiple delivery options in a single purchase.</td>
```

    </tr>
    <tr>
      <td colspan="2" valign="top">
        <strong id="orderformconfiguration.allowmanualprice">allowManualPrice</strong>
      </td>
      <td valign="top">
        Boolean
      </td>
      <td>Allows the editing of SKU prices directly in the cart.</td>
    </tr>
    <tr>
      <td colspan="2" valign="top">
        <strong id="orderformconfiguration.maxintofwhitelabelsellers">maxIntOfWhiteLabelSellers</strong>
      </td>
      <td valign="top">
        Int
      </td>

>       <td>Allows the input of a limit of white label sellers involved on the cart.</td>

Reviewer: ```suggestion
      <td>Maximum number of white-label sellers allowed in the cart.</td>
```

    </tr>
    <tr>
      <td colspan="2" valign="top">
        <strong id="orderformconfiguration.maskfirstpurchasedata">maskFirstPurchaseData</strong>
      </td>
      <td valign="top">
        Boolean
      </td>

>       <td>Allows, on a first purchase, masking client's data. It could be useful when a shared cart is used and the client doesn't want to share its data.</td>

Reviewer: ```suggestion
      <td>Masks customer data for first-time purchases. It could be useful when a shared cart is used, and the client doesn't want to share its data.</td>
```
